examples/pull.rs

```diff
@@ -0,0 +1,200 @@
+/*
+ * libgit2 "pull" example - shows how to pull remote data into a local branch.
+ *
+ * Written by the libgit2 contributors
+ *
+ * To the extent possible under law, the author(s) have dedicated all copyright
+ * and related and neighboring rights to this software to the public domain
+ * worldwide. This software is distributed without any warranty.
+ *
+ * You should have received a copy of the CC0 Public Domain Dedication along
+ * with this software. If not, see
+ * <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+extern crate docopt;
+extern crate git2;
+#[macro_use]
+extern crate serde_derive;
```

Comment:
When you rebase on master, the project has been updated to 2018 edition, so you can drop these `extern crate` lines.  Will still need a `use serde_derive::Deserialize;`, though.

Reply:
I'll do that at the end so I only have to take the hit once. :-D